service/topology_coordinator.cc

```diff
@@ -740,6 +755,82 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
     }
 
+    // If a node crashes after initiating gossip and before joining group0, it becomes orphan and 
+    // remains in gossip. This backgound fiber periodically checks for such nodes and purges those 
+    // entries from gossiper by commiting the node as left in raft.
+    future<> gossiper_orphan_remover_fiber() {
+        rtlogger.debug("start gossiper orphan remover fiber");
+        bool do_speedup_fiber = false;
+        co_await utils::get_local_injector().inject("fast_orphan_removal_fiber", [&](auto& handler) -> future<> {
+            do_speedup_fiber = true;
+            co_return;
+        });
+        while (!_as.abort_requested()) {
+            try {
+                auto guard = co_await start_operation();
+                std::vector<canonical_mutation> updates;
+                std::optional<gms::inet_address> orphan_ip;
+                co_await utils::get_local_injector().inject("speedup_orphan_removal", [&](auto& handler) -> future<> {
+                    auto fetched_orphan_ip = handler.get("orphan_ip");
+                    orphan_ip = gms::inet_address(sstring{*fetched_orphan_ip});
+                    co_return;
+                });
+                std::string reason = ::format("Ban the orphan nodes in group0. Orphan nodes HostId/IP:");
+                _gossiper.for_each_endpoint_state([&](const gms::inet_address& addr, const gms::endpoint_state& eps) -> void {
+                    auto current_timestamp =
+                            std::chrono::duration_cast<std::chrono::seconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
```

Comment:
Do I understand correctly that you're casting it to `seconds` because `generation` also uses seconds resolution? Maybe that latter fact deserves a comment

Reply:
`time_since_epoch` returns nanoseconds, and I converted it to seconds, then to int32_t using `count()`. And yes `generation ` also uses seconds. At last we calculate difference and compare with timeout(int32_t)

Added comment